### DISCUSSION TITLE
Is there any way to debug access violations?

### DISCUSSION THREAD

**Discussion by Minding000** [bytedeco/javacpp-presets#1387] [Q&A] [answered]
I am using the LLVM library in Kotlin (JVM) and am very grateful this repository exists.
But I find myself running into access violations every now and then (due to mistakes in my code) and they are very hard to debug (only by trial and error really).

Is there some way to debug the cause of `EXCEPTION_ACCESS_VIOLATION` (printing the C++ stack trace for example) or to prevent them?
That would improve the usability a lot :)

**Comment by saudet** [comment, marked as answer]
That's what tools like Valgrind and AddressSanitizer are for:
 - https://valgrind.org/
 - https://clang.llvm.org/docs/AddressSanitizer.html
